pkg/provider/kubernetes/ingress/kubernetes.go

```diff
@@ -313,53 +313,57 @@ func (p *Provider) loadConfigurationFromIngresses(ctx context.Context, client Cl
 				conf.HTTP.Services["default-backend"] = service
 			}
 		}
+
 		for _, rule := range ingress.Spec.Rules {
 			if err := checkStringQuoteValidity(rule.Host); err != nil {
 				log.FromContext(ctx).Errorf("Invalid syntax for host: %s", rule.Host)
 				continue
 			}
 
-			for _, p := range rule.HTTP.Paths {
-				service, err := loadService(client, ingress.Namespace, p.Backend)
-				if err != nil {
-					log.FromContext(ctx).
-						WithField("serviceName", p.Backend.ServiceName).
-						WithField("servicePort", p.Backend.ServicePort.String()).
-						Errorf("Cannot create service: %v", err)
-					continue
-				}
-
-				if err = checkStringQuoteValidity(p.Path); err != nil {
-					log.FromContext(ctx).Errorf("Invalid syntax for path: %s", p.Path)
-					continue
-				}
+			if rule.HTTP != nil {
```

Comment:
why do we need to get in p.updateIngressStatus if rule.HTTP is nil?

Reply:
The problem is the code right now assumes every ingress has a non-nil `rule.HTTP` which causes it to stack trace.  This just avoid that's for scenarios either other ingresses like `ingress-nginx` where you can have an ingress with an annotation for a redirect but still have a host to make it work.

I can provide said stack trace if you're interested in it.  That said, The `Ingress` I added as a test fixture is basically what I have setup minus the `ingress-nginx`  annotation.

Reply:
Oh I understand your PR and the motivation for it. I'm just asking @ldez why he changed your more readable "continue early", to a less readable inverted check. The only reason I see for that, is that we would still want to get into p.updateIngressStatus even when rule.HTTP is nil but I don't know why we would want that.

Reply:
Ah gotcha, I was curious on that as well, but not being very familiar with the style of the project just let it go :)

Reply:
>why do we need to get in p.updateIngressStatus if rule.HTTP is nil?

Because the systems that rely on ingress status often do not require or interact with `rule.HTTP`, and may continue to function as normal.

Reply:
ok, but I was hoping we knew exactly what would break if we just did

    if rule.HTTP == nil {
      continue
    }

instead. but it's not essential, so I'll figure it out myself later.